### DISCUSSION TITLE
Upgrade to new Spring 3.0.6 and Jedis 5.0.0

### DISCUSSION THREAD

**Discussion by paulmac** [medovarszki/ibkrfacade#5] [Ideas]
I've done this for my own purposes but wonder if you'd like to have me commit them or make a fork. I found some issues with swagger also, quite a number of small changes. Next step might be to upgrage to Jedis-OM

1. My aim is to continue this work using webhooks from TradingView as trade signals for a particular Strategy. Forward Testing of Startegies can be performed by including a repository of Trades and Results

**Comment by medovarszki** [comment]
Hey! I'm glad you find this project interesting! Any contributions are welcomed, just send a pull request with your changes :)

**Comment by paulmac** [comment]
I am now hooking up another API  for TradingView as Alerts can be issued
with a WebHook.

I have so far an Alert and Strategy pojo which serve to forward test any
strategy published on TradingView.

TradingView uses PineScript which is a pseudo language for trading but
ultimately I'd like to use chatGPT to convert profitable strategies to
Python with QuantConnect

Integrating Redis-OM for this new interface.
https://github.com/redis/redis-om-spring/blob/main/README.md

I think you made an excellent choice with Redis, those guys are working
hard to keep their noses in front in that market

Regards

Paul

*Tel* +61 421227775


On Sun, May 14, 2023 at 11:44 PM Mihaly Medovarszki <
***@***.***> wrote:

> Hey! I'm glad you find this project interesting! Any contributions are
> welcomed, just send a pull request with your changes :)
>

will do

**Reply by paulmac** [reply]
Hi Mihaly, I've commited a whole bunch of changes of ibkr to my fork. It is rearranged for Redis-OM style. Read the comments and have a look at the code and see if you agree with what I have done. I do not not use the ContractDetail, it's to cumbersome, I have a class called SimpleContract which contains the 4 or 5 attributes necesary, it is a Redis OM Json object, I could include it and merge with your Option, making it a Json object also if you agree, also there needs to be a client now to implement the ExecHandler interaface for TWs responses. The rest of the changes should be logical for you. Paul.
